drake/systems/plants/collision/FCLModel.cpp

```diff
@@ -0,0 +1,421 @@
+#include <iostream>
+
+//#include "drake/systems/plants/collision/DrakeCollision.h"
+#include "FCLModel.h"
+//#include "fcl/shape/geometric_shapes.h"
+//#include "fcl/shape/geometric_shape_to_BVH_model.h"
+
+#include "fcl/shape/geometric_shapes.h"
+#include "fcl/collision_data.h"
+#include "fcl/distance.h"
+#include "fcl/collision.h"
+#include "fcl/BVH/BVH_model.h"
+#include "fcl/shape/geometric_shape_to_BVH_model.h"
+
+#include "drake/systems/plants/collision/DrakeCollision.h"
+
+using namespace std;
+using namespace fcl;
+using namespace Eigen;
+
+namespace DrakeCollision {
+
+BVHModel<OBBRSS>* FCLModel::newFCLBoxShape(const DrakeShapes::Box& geometry,
+                                           bool use_margins) {
+  // NOTE:  Ignore margins for now.
+  use_margins = use_margins;
+
+  fcl::Box shape(geometry.size(0), geometry.size(1), geometry.size(2));
+  fcl::BVHModel<OBBRSS>* bvh_shape = new fcl::BVHModel<OBBRSS>();
+  fcl::generateBVHModel(*bvh_shape, shape, Transform3f());
+
+  return bvh_shape;
+}
+
+BVHModel<OBBRSS>* FCLModel::newFCLSphereShape(
+    const DrakeShapes::Sphere& geometry, bool use_margins) {
+  // NOTE:  Ignore margins for now.
+  use_margins = use_margins;
+
+  fcl::Sphere shape(geometry.radius);
+  fcl::BVHModel<OBBRSS>* bvh_shape = new fcl::BVHModel<OBBRSS>();
+  int numLong = 10;
+  int numLat = 8;
+  fcl::generateBVHModel(*bvh_shape, shape, Transform3f(), numLong, numLat);
+
+  return bvh_shape;
+}
+
+BVHModel<OBBRSS>* FCLModel::newFCLCylinderShape(
+    const DrakeShapes::Cylinder& geometry, bool use_margins) {
+  // NOTE:  Ignore margins for now.
+  use_margins = use_margins;
+
+  fcl::Cylinder shape(geometry.radius, geometry.length);
+  fcl::BVHModel<OBBRSS>* bvh_shape = new fcl::BVHModel<OBBRSS>();
+  int numLength = 1;
+  int numTheta = 10;
+  fcl::generateBVHModel(*bvh_shape, shape, Transform3f(), numTheta, numLength);
+
+  return bvh_shape;
+}
+
+BVHModel<OBBRSS>* FCLModel::newFCLCapsuleShape(
+    const DrakeShapes::Capsule& geometry, bool use_margins) {
+  // NOTE:  Ignore margins for now.
+  use_margins = use_margins;
+
+  fcl::Capsule shape(geometry.radius, geometry.length);
+  fcl::BVHModel<OBBRSS>* bvh_shape = new fcl::BVHModel<OBBRSS>();
+
+  // fcl does not have a method to generate a capsule mesh.
+  // TODO: Make one.
+  // fcl::generateBVHModel(*bvh_shape, shape, Transform3f(), numTheta,
+  // numLength);
+
+  return bvh_shape;
+}
+
+// TODO: Figure out how to convert mesh verts to fcl mesh.
+BVHModel<OBBRSS>* FCLModel::newFCLMeshShape(const DrakeShapes::Mesh& geometry,
```

Comment:
Drake's implementation of mesh shapes seems to be using convex hulls for this in [BulletModel::newBulletMeshShape](https://github.com/RobotLocomotion/drake/blob/master/drake/systems/plants/collision/BulletModel.cpp#L118). Maybe we'd like to do that here as well for now?